tests/unit/test_messages_printer.py

```diff
@@ -232,6 +232,170 @@ def test_writeline_having_previous_message_ephemeral(capsys, monkeypatch, log_fi
     assert not err
 
 
+# -- tests for the writing bar function
+
+
+def test_writebar_simple(capsys, monkeypatch, log_filepath):
+    """Complete verification of _write_line for a simple case."""
+    monkeypatch.setattr(messages, "_get_terminal_width", lambda: 40)
+    printer = _Printer(log_filepath)
+
+    msg = _MessageInfo(sys.stdout, "test text", bar_progress=50, bar_total=100)
+    printer._write_bar(msg)
+    assert printer.unfinished_stream == sys.stdout
+
+    out, err = capsys.readouterr()
+    assert not err
+
+    # output completes the terminal width (leaving space for the cursor), and
+    # without a finishing newline
+    assert len(out) == 39
+    assert out == "test text [##########          ] 50/100"
+
+
+def test_writebar_simple_empty(capsys, monkeypatch, log_filepath):
+    """The indicated progress is zero."""
+    monkeypatch.setattr(messages, "_get_terminal_width", lambda: 40)
+    printer = _Printer(log_filepath)
+
+    msg = _MessageInfo(sys.stdout, "test text", bar_progress=0, bar_total=100)
+    printer._write_bar(msg)
+
+    out, _ = capsys.readouterr()
+    assert len(out) == 39
+    assert out == "test text [                     ] 0/100"
+
+
+def test_writebar_simple_total(capsys, monkeypatch, log_filepath):
+    """The indicated progress is the total."""
+    monkeypatch.setattr(messages, "_get_terminal_width", lambda: 40)
+    printer = _Printer(log_filepath)
+
+    msg = _MessageInfo(sys.stdout, "test text", bar_progress=100, bar_total=100)
+    printer._write_bar(msg)
+
+    out, _ = capsys.readouterr()
+    assert len(out) == 39
+    assert out == "test text [###################] 100/100"
+
+
+def test_writebar_simple_exceeding(capsys, monkeypatch, log_filepath):
+    """The indicated progress exceeds the total."""
+    monkeypatch.setattr(messages, "_get_terminal_width", lambda: 40)
+    printer = _Printer(log_filepath)
+
+    msg = _MessageInfo(sys.stdout, "test text", bar_progress=120, bar_total=100)
+    printer._write_bar(msg)
+
+    out, _ = capsys.readouterr()
+    assert len(out) == 39
+    assert out == "test text [###################] 120/100"
```

Comment:
Not related to this test, but will the output look correct if `bar_progress` is set to something like `1000/100`, and then back to `120/100`?

Reply:
Yes. Each time the line is drawn, it uses all the terminal's width.